plugin/lighthouse/handler.go

```diff
@@ -0,0 +1,61 @@
+package lighthouse
+
+import (
+	"context"
+	"net"
+	"strings"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/nonwriter"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+)
+
+// ServeDNS implements the plugin.Handler interface.
+func (lh *Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	state := request.Request{W: w, Req: r}
+
+	a := new(dns.Msg)
+	a.SetReply(r)
+	a.Authoritative = true
+
+	services := lh.SvcsMap
+	svcName := strings.Split(state.QName(), ".")[0]
+	ip := services[svcName]
+	if ip == "" {
+		return onFailure(lh, ctx, w, r)
+	}
+	rr := new(dns.A)
+
+	switch state.Family() {
+	case 1:
+		rr.Hdr = dns.RR_Header{Name: state.QName(), Rrtype: dns.TypeA, Class: state.QClass()}
+		rr.A = net.ParseIP(ip).To4()
+	default:
+		log.Debugf("IPv6 queries not supported yet")
+		return onFailure(lh, ctx, w, r)
+	}
+
+	a.Answer = []dns.RR{rr}
+
+	log.Debugf("Responding to query with '%s'", a.Answer)
+	if w.WriteMsg(a) != nil {
+		log.Debugf("Failed to writeMsg")
```

Comment:
`WriteMsg` returns an error - we should log it here along with the context, eg

 `log.Debugf("Failed to write message %#v: %v", a, err)`

Also maybe this should be Error or Warning?

Reply:
Hmm... most other plugins just ignore coz this failing is like printf() failing. Added a check just to satisfy golint. In case of failure, we will go to next plugin and if all fails return NXDOMAIN to user.

Reply:
Shouldn’t the plugin return `SERVFAIL` in this case?

Reply:
yeah I don't know what circumstance(s) would cause this to fail or why other plugins ignore the return error - maybe it's oversight or seen as an error that would never occur or just not following best practices.  But if we're going to log it then I think it's useful to include as much context as we can. Since this is an error condition and I assume would be uncommon, it feels like an Error or Warning log level, ie useful to emit by default.

Reply:
Done.